src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1577,31 +1577,28 @@ public static uint GetNextModuleId()
             return ++s_nextModuleId;
         }
 
-        public static ProtocolMessages.Module ConvertToModule(MODULE_INFO debugModuleInfo)
+        public static ProtocolMessages.Module ConvertToModule(MODULE_INFO debugModuleInfo, IDebugModule2 module)
         {
             var mod = new ProtocolMessages.Module(GetNextModuleId(), debugModuleInfo.m_bstrName)
             {
                 Path = debugModuleInfo.m_bstrUrl,
-                VsTimestampUTC = (debugModuleInfo.dwValidFields & enum_MODULE_INFO_FIELDS.MIF_TIMESTAMP) != 0 ? FileTimeToPosix(debugModuleInfo.m_TimeStamp).ToString(CultureInfo.InvariantCulture) : "",
+                VsTimestampUTC = (debugModuleInfo.dwValidFields & enum_MODULE_INFO_FIELDS.MIF_TIMESTAMP) != 0 ? FileTimeToPosix(debugModuleInfo.m_TimeStamp).ToString(CultureInfo.InvariantCulture) : null,
                 Version = debugModuleInfo.m_bstrVersion,
                 VsLoadAddress = debugModuleInfo.m_addrLoadAddress.ToString(CultureInfo.InvariantCulture),
                 VsPreferredLoadAddress = debugModuleInfo.m_addrPreferredLoadAddress.ToString(CultureInfo.InvariantCulture),
                 VsModuleSize = (int)debugModuleInfo.m_dwSize,
                 VsLoadOrder = (int)debugModuleInfo.m_dwLoadOrder,
                 SymbolFilePath = debugModuleInfo.m_bstrUrlSymbolLocation,
                 SymbolStatus = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_SYMBOLS) != 0 ? "✓" : "✗",
-                VsIs64Bit = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_64BIT) != 0
-                // IsOptimized = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_OPTIMIZED) != 0 ? true : ((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_UNOPTIMIZED) != 0 ? false : null) // not set by gdb
+                VsIs64Bit = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_64BIT) != 0,
+                IsOptimized = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_OPTIMIZED) != 0 ? true : ((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_UNOPTIMIZED) != 0 ? false : null) // not set by gdb
                 // test -- need to delete?
                 // IsUserCode = (debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_SYSTEM) != 0 // not set by gdb
             };
-            if((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_OPTIMIZED) != 0)
-            {
-                mod.IsOptimized = true;
-            } else if ((debugModuleInfo.m_dwModuleFlags & enum_MODULE_FLAGS.MODULE_FLAG_UNOPTIMIZED) != 0)
-            {
-                mod.IsOptimized = false;
-            }
+
+            var status = (module as IDebugModule3).IsUserCode(out int isUserCode);
```

Comment:
If somehow `module` is not a `IDebugModule3`, the result would be `null` and you would be calling `IsUserCode` on `null`.

Create a new variable and null check it.

Reply:
You can fix this with: 

```C#
if (module is IDebugModule3 module3 &&
    module3.IsUserCode(out int isUserCode) == VSConstants.S_OK)
{
    ...
}
```

---
In reply to: [503486210](https://github.com/microsoft/MIEngine/pull/1054#discussion_r503486210) [](ancestors = 503486210)